packages/atomic/src/components/search/result-template-components/atomic-result-text/atomic-result-text.tsx

```diff
@@ -102,15 +102,18 @@ export class AtomicResultText implements InitializableComponent {
     }
 
     const textValue = `${resultValueAsString}`;
+
+    if (!this.shouldHighlight) {
+      return getFieldValueCaption(this.field, textValue, this.bindings.i18n);
+    }
+
     const highlightsValue = ResultTemplatesHelpers.getResultProperty(
       this.result,
       `${this.field}Highlights`
     ) as HighlightUtils.HighlightKeyword[];
 
-    if (this.shouldHighlight && highlightsValue) {
+    if (highlightsValue) {
       return this.renderWithHighlights(textValue, highlightsValue);
     }
-
-    return getFieldValueCaption(this.field, textValue, this.bindings.i18n);
   }
```

Comment:
Since this code was still trying to retrieve the highlights value even if shouldHighlight was false, this was causing an error when used with products. Do we want to instead add a `atomic-product-text` template component and keep this one as it was previously?

Reply:
Rule of thumb should be: we must not use any components that are under the `/search` folder in anything related to commerce. It's more than a rule of thumb, it's 100% mandatory and must not slip 😂 

We definitely don't want to have those kind of cross pollution because it creates issues like this: We must not have any exception/if/different branching in a top level web component: The strategy is to have very specific/dedicated components which caters to only one use case at a time. Going down that route === definitely going to introduce regression for other clients using atomic in production.

The code reuse needs to be done with functional components/utility function when there's a lot of overlap.


Going with my other components about the templates themselves in the local sample pages, I would prefer if they stayed ugly until we implement all the commerce specific product list components that are marked in the epic.

Reply:
reverted this change